### ISSUE TITLE
Use same tense in changelog messages

### ISSUE THREAD

**Issue by theborakompanioni** [olivierlacan/keep-a-changelog#54]
I opened this issue because I do not quite know what to do. I have rather often read to use the "imperative present tense" in commit messages. What about changelogs? The proposed special words like "added" and "removed" are all in past tense. The main example mixes them.

Present tense:

```
### Fixed
- Fix typos in recent README changes.
- Update outdated unreleased diff link.
```

Past tense:

```
### Added
- Unreleased section to gather unreleased changes and encourage note
keeping prior to releases.
### Changed
- Clarified the section on "Is there a standard change log format?".
```

I suggest to be consistent about the style - but I do not know which one to prefer (to be fair: I have a preference, but I want to hear other opinios before submitting a pull request). Any thoughts?

**Comment by lunemec** [comment]
Well I'd suggest the past tense, it falls into the conversation:
    What did you do in this version?
    Developer: I fixed this, broke this, added this .. etc ..

**Comment by olivierlacan** [comment]
The trick is that it's recommended to [write commit messages in imperative present tense](http://git-scm.com/book/ch5-2.html#Commit-Guidelines). This means that if we settle on change logs using the past tense people will have to transpose good commit titles. I'm not sure that's a good idea.

That said, I did make the category titles past tense because it seems logical to talk about a release as something that affects the past, not the present, as a commit message does.

**Comment by elyscape** [comment]
> This means that if we settle on change logs using the past tense people will have to transpose good commit titles. I'm not sure that's a good idea.

I think it's fine. People shouldn't just be dumping their commit titles into the changelog, after all.

**Comment by theborakompanioni** [comment]
I’m still trying to wrap my head around it..  I have only small projects with fast release cycles - so my change logs are rather short. I am mixing tenses at the moment and I have to say: I am quite fine with that.
I cannot really argue for larger projects where change logs tend to be multiple pages long and should be consistent and searchable. Personally, I tend to use past tense because like commit messages should "express what the changes do when you apply them", a changelog message should express that you really *did* apply them.

**Comment by elyscape** [comment]
I think the items in `Added`, `Deprecated`, `Removed`, and `Fixed` should be nouns, and `Changed` and `Security` should be present-tense descriptions of the changes or security notes.  For example (adapated from [the change log for hiera-eyaml](https://github.com/TomPoulton/hiera-eyaml/blob/master/CHANGES.md)):
```markdown
## 2.0.4 - 2014-11-24
### Added
- Change log
- Initial support for spaces in filenames
- Preamble to edit mode to make it easier to remember how to edit
- Colorized output
### Changed
- Temp files now have `.yaml` extension to help editors provide syntax highlighting
- Debug options are more useful
### Fixed
- Various documentation
### Security
- Edit mode no longer leaks sensitive info to stdin
  (example security fix added for demonstration; not actually from the project)
```
My reasoning is that `Added`, `Deprecated`, `Removed`, and `Fixed` are things that were added, deprecated, removed, or fixed.

**Comment by greg0ire** [comment]
I would recommend @elyscape : solution, but with past tense for the `Security` and `Changed` items since 

> a changelog message should express that you really did apply them.

(from @theborakompanioni)

Can we agree on something ?

**Comment by Soullivaneuh** [comment]
Personally I agree with @olivierlacan https://github.com/olivierlacan/keep-a-changelog/issues/54#issuecomment-74681533

**Comment by greg0ire** [comment]
> Personally I agree with @olivierlacan #54 (comment)

Do you mean you would do this : 

```
# Added
- add oranges
- add apples
```

@olivierlacan main point is that if you use present tense, you can reuse good commit messages. But keepachangelog.com explicitely states that you cannot the changelog is not a subset of the git log (if it is, what good is it?). This sentence in particular show that you can't just do copy / paste : 

> As is the difference between good comments and the code itself, so is the difference between a change log and the commit log: one describes the why, the other the how.

**Comment by Soullivaneuh** [comment]
@greg0ire about present tense, not necessary "Add ...".

**Comment by contivero** [comment]
I'm not sure if it's the right place, or I should open another issue about it, but along the lines of the discussion, does anybody else find repeating the verb redundant?

As in the snippet by @greg0ire, instead of: `# Added`, `add ...`, `add ...`, isn't it better to assume it and list directly what was added? I.e. 
```
# Added
- oranges
- apples
```
That's exactly what @elyscape's example does btw.

**Comment by greg0ire** [comment]
I fully agree with you @contivero , see for instance https://github.com/sonata-project/SonataCacheBundle/pull/159

**Comment by lunemec** [comment]
This question is being discussed for 2 years now. Ehhhh.... maybe close?

**Comment by olivierlacan** [comment]
@lunemec What's the rush? 😄

**Comment by olivierlacan** [comment]
But okay, let me clarify my thoughts as they've evolved from the discussion here and in outside.

I prefer the past tense (`Added`, `Removed`, `Changed`, `Deprecated`) because we're talking about something that is now done and out in the world. It's clearly not the same thing as a commit message which describe the purpose of a unit of work. The changelog entries describe *what* the release brought to the world.

This might be a lovely addition to make in the 1.1 release of Keep a Changelog.

PS: @contivero please do open an issue about that, that's a great idea.

**Comment by olivierlacan** [comment]
Here's a little draft: 
![image](https://user-images.githubusercontent.com/65950/36784842-2eff3f74-1c81-11e8-8437-053398cf4acc.png)

**Comment by theborakompanioni** [comment]
Having thought about it long enough and coming to the conclusion that every project might use it's own style. The only thing that matters is to keep a changelog in the first place and trying to be consistent, regardless of the tense you use. If you keep a changelog, you are doing the "right thing"™.  :muscle:

Don't worry too much.
I am closing this issue now.
